src/visualization/parsers/boxmodel.ts

```diff
@@ -0,0 +1,186 @@
+/// <reference path="../../../node_modules/clarity-js/clarity.d.ts" />
+import { IParser } from "../components/Snapshot";
+
+export default class BoxModel implements IParser {
+    private layouts: { [index: number]: Node } = {};
+    private states: { [index: number]: IElementLayoutState } = {};
+    private document: Document;
+    private base: string;
+   
+    setup(document: Document, frame: HTMLIFrameElement, base: string, thumbnail? : boolean) {
+        this.layouts = {};
+        this.document = document;
+        this.base = base;
+    }
+
+    private domInsert(node: Node, parent?: Node, nextSibling?: Node) {
+        if (parent) {
+            if (nextSibling && nextSibling.parentNode === parent) {
+                nextSibling.parentNode.insertBefore(node, nextSibling);
+            }
+            else {
+                try {
+                    parent.appendChild(node);
+                }
+                catch (ex) {
+                    console.warn("Error while inserting a node: " + ex);
+                }
+            }
+            return node;
+        }
+        return null;
+    }
+
+    private domRemove(node: Node) {
+        if (node && node.parentNode) {
+            node.parentNode.removeChild(node);
+        }
+        else {
+            console.warn(`Remove: Cannot remove ${node}`);
```

Comment:
>console.warn(`Remove: Cannot remove ${node}`); [](start = 12, length = 46)

nit: stylistically we should be consistent with if we are using interpolation

Reply:
Made it consistent.